tracer/src/Datadog.Trace/OTelMetrics/OtlpExporter.cs

```diff
@@ -0,0 +1,275 @@
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Telemetry;
+
+#nullable enable
+
+namespace Datadog.Trace.OTelMetrics
+{
+    /// <summary>
+    /// OTLP Exporter implementation that exports metrics using the OpenTelemetry Protocol.
+    /// This is the concrete implementation of the MetricExporter base class that handles OTLP protocol specifics.
+    /// Supports both grpc and http/protobuf transports as specified in the RFC.
+    /// This is a Push Metric Exporter that sends metrics via the OpenTelemetry Protocol.
+    /// </summary>
+    internal class OtlpExporter(Configuration.TracerSettings settings) : MetricExporter
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(OtlpExporter));
+        private readonly HttpClient _httpClient = new();
+        private readonly Configuration.TracerSettings _settings = settings ?? throw new ArgumentNullException(nameof(settings));
+
+        /// <summary>
+        /// Exports a batch of metrics using OTLP protocol.
+        /// This method implements the MetricExporter base class Export operation.
+        /// </summary>
+        /// <param name="metrics">Batch of metrics to export</param>
+        /// <returns>ExportResult indicating success or failure</returns>
+        public override ExportResult Export(IReadOnlyList<MetricPoint> metrics)
+        {
+            // For backward compatibility, call the async version synchronously
+            return ExportAsync(metrics).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Exports a batch of metrics using OTLP protocol asynchronously.
+        /// This is the preferred method for better performance.
+        /// </summary>
+        /// <param name="metrics">Batch of metrics to export</param>
+        /// <returns>ExportResult indicating success or failure</returns>
+        public override async Task<ExportResult> ExportAsync(IReadOnlyList<MetricPoint> metrics)
+        {
+            if (metrics.Count == 0)
+            {
+                Log.Debug("No metrics to export.");
+                return ExportResult.Success; // Success - nothing to do
+            }
+
+            // Record export attempt
+            TelemetryFactory.Metrics.RecordCountMetricsExportAttempts(GetProtocolTag());
+
+            try
+            {
+                // Take snapshots of all metrics for export
+                var metricPoints = new List<MetricPoint>();
+                foreach (var metric in metrics)
+                {
+                    metric.TakeSnapshot(outputDelta: _settings.OtlpMetricsTemporalityPreference == Configuration.OtlpTemporality.Delta);
+                    metricPoints.Add(metric);
+                }
+
+                // Serialize to OTLP protobuf format
+                var otlpPayload = OtlpMetricsSerializer.SerializeMetrics(metricPoints, _settings);
```

Comment:
I see a lot of the `_settings` being passed around through the serialization. Since you passed the `TracerSettings` to this `OtlpExporter` instance, I also recommend creating an individual instance of the `OtlpMetricsSerializer` class, that way you can prepopulate the serializer with all of the information you need (like the Resources, which you should only need to change if settings change). So in general, see if you can make this entire sequence of calls use object instances, and pre-populate as much as you can so you don't have to spend valuable CPU re-calculating things during the Export call

Reply:
I tried applying this one in the last commit: https://github.com/DataDog/dd-trace-dotnet/pull/7514/commits/bd07ebf7b238c74551b3693267294b1a9255fd87